nebuchadnezzar/nebu/utils.py

```diff
@@ -34,3 +34,47 @@ def re_first_or_default(
 ) -> Optional[str]:
     match = re.search(pattern, s)
     return match.group(0) if match is not None else default
+
+
+def merge_by_index(lhs, rhs, merge_sequence, default):
+    return list(
+        recursive_merge(
+            dict(enumerate(lhs)),
+            dict(enumerate(rhs)),
+            merge_sequence=merge_sequence,
+            default=default,
+        ).values()
+    )
+
+
+def recursive_merge(
+    lhs,
+    rhs,
+    *,
+    merge_sequence=merge_by_index,
+    default=lambda lhs, rhs: rhs if rhs is not None else lhs
```

Comment:
Could this error if both the left and right values are defined?

That would catch cases like "Why isn't the answer showing up? I put the answer in the (public) JSON but it still doesn't show up (because it's also defined in the private JSON)"

(related to the [same code in CAT](https://github.com/openstax/vscode-h5p/pull/13/files#r1541948294))